tests/src/cppstd/hipInvocable.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (c) 2015 - present Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+/* HIT_START
+ * BUILD: %t %s ../test_common.cpp LINK_OPTIONS hiprtc EXCLUDE_HIP_PLATFORM nvcc
+ * TEST: %t
+ * HIT_END
+ */
+
+#include <hip/hip_runtime.h>
+#include <test_common.h>
+
+struct CallableStruct {
+    double operator()() { return 1.0; }
+    double operator()(int a) { return a * 1.0; }
+};
+
+int main() {
+    bool isCallable1 = hip_impl::is_callable<CallableStruct()>::type();
+    bool isCallable2 = hip_impl::is_callable<CallableStruct(int)>::type();
+    std::conditional<hip_impl::is_callable<CallableStruct(int)>{}, char, long>::type Type;
+    if (typeid(Type).name() != typeid(char).name()) {
```

Comment:
Why not just write `if(hip_impl::is_callable<CallableStruct(int)>{})`? Its much simpler.